include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,254 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_access_traits_h
+#define dealii_arborx_access_traits_h
+
+#include <deal.II/base/config.h>
+
+#if defined(DEAL_II_WITH_ARBORX) && defined(DEAL_II_HAVE_CXX17)
+#  include <deal.II/base/bounding_box.h>
+
+#  include <ArborX.hpp>
+
+namespace ArborX
+{
+  /**
+   * This struct allows ArborX to use std::vector<dealii::BoundingBox> as
+   * primitive.
+   */
+  template <int dim, typename Number>
+  struct AccessTraits<std::vector<dealii::BoundingBox<dim, Number>>,
+                      PrimitivesTag>
+  {
+    static_assert(dim != 1, "dim equal to one is not supported.");
+
+    /**
+     * Return the size of the vector @p v.
+     */
+    static std::size_t
+    size(std::vector<dealii::BoundingBox<dim, Number>> const &v)
+    {
+      return v.size();
+    }
+
+    /**
+     * Return an ArborX::Box from the dealii::BoundingBox v[i].
+     */
+    static Box
+    get(std::vector<dealii::BoundingBox<dim, Number>> const &v, std::size_t i)
+    {
+      auto const &[min_corner, max_corner] = v[i].get_boundary_points();
```

Comment:
I think it's easier if we don't require `C++17`. `Point` objects are not expensive to copy and they need to be copied to construct `Box` anyway.